admin/webapp/websrc/app/routes/compliance/compliance.component.ts

```diff
@@ -40,21 +35,9 @@ export class ComplianceComponent implements OnInit, OnDestroy {
     private complianceService: ComplianceService,
     private complianceCsvService: ComplianceCsvService,
     public complianceFilterService: ComplianceFilterService,
-    private multiClusterService: MultiClusterService
   ) {}
 
-  ngOnInit(): void {
-    this._switchClusterSubscription =
-      this.multiClusterService.onClusterSwitchedEvent$.subscribe(() => {
-        this.refresh();
-      });
-  }
-
-  ngOnDestroy(): void {
-    if (this._switchClusterSubscription) {
-      this._switchClusterSubscription.unsubscribe();
-    }
-  }
+  ngOnInit(): void {}
```

Comment:
Yes. I planned to do refresh for all by cluster triggered by header to let the component processes onInit() again, it's because the permission changes will influence the appearance of the components, like table shows the the edit field with the editing button for write user on but not when the permission is changed to only view right, and it depends on the more complete loading from onInit() rather than the partial data refresh.

header.component.ts ( use onClusterSwitchedEvent$ and router to trigger the complete refresh())
```
this.multiClusterService.onClusterSwitchedEvent$.subscribe(() => {
        const currentUrl = this.router.url;
        this.router
          .navigateByUrl('/', { skipLocationChange: true })
          .then(() => {
            this.router.navigate([currentUrl]);
          });
      });
```

onGetClustersFinishEvent$ is for some components which have extra actions after refresh complete.

For instance, SAML setting page needs to redirect to /setting page's entrance after cluster changed based on the original processes. 
```
this._getClustersFinishSubscription =
      this.multiClusterService.onGetClustersFinishEvent$.subscribe(data => {
        this.router.navigate(['settings']);
      });
```